[DOCS] Cleans up monitoring configuration steps

lcawl opened the pull request:
This PR is related to https://github.com/elastic/stack-docs/pull/138 and https://github.com/elastic/kibana/pull/24881 

It adds information that previously existed in https://www.elastic.co/guide/en/elastic-stack-overview/master/monitoring-production.html and removes information that is more appropriate for the Kibana documentation. 

It also cleans up some of the "X-Pack" terminology.

elasticmachine commented:
Pinging @elastic/es-core-infra

lcawl commented:
You can preview these changes here:
https://doc-review-63bf6.firebaseapp.com/es2/configuring-monitoring.html
https://doc-review-63bf6.firebaseapp.com/es2/es-monitoring.html

ycombinator commented on docs/reference/monitoring/configuring-monitoring.asciidoc:
```diff
@@ -53,31 +53,40 @@ patterns. For example, to include all indices that start with `test` except
 
-Alternatively, you can use an `http` exporter to send data to a separate 
-monitoring cluster. See <<http-exporter>>. 
+Alternatively, you can use <<http-exporter,`http` exporters>> to send data to a 
```
The same discussion applies to the "local exporters" section above.

lcawl replied:
Thanks for the clarification. I was actually thinking that there were exporters on each node, which is why I made it plural, but I agree it's less confusing to just make it singular.

ycombinator commented on docs/reference/monitoring/index.asciidoc:
```diff
@@ -5,38 +5,42 @@
+
+beta[] Alternatively, in 6.4 and later, you can use {metricbeat} to collect and 
+ship data about {kib}, rather than routing it through {es}. In 6.5 and later, you 
+can also use {metricbeat} to collect and ship data about {es}. 
```
For the metricbeat alternative, I think it would be useful to convey that metricbeat ships directly to the monitoring cluster. So what about changing the first sentence to something like this?

> Alternatively, in 6.4 and later you can use {metricbeat} to collect monitoring data about {kib} and ship it directly to the monitoring {es} cluster rather than routing it through the production {es} cluster.

Obviously, feel free to edit as you see fit!

lcawl commented:
Thanks @ycombinator, I've addressed your feedback.  I also added examples of how to update the cluster-wide settings in https://doc-review-63bf6.firebaseapp.com/es2/configuring-monitoring.html

ycombinator commented on docs/reference/monitoring/configuring-monitoring.asciidoc:
```diff
@@ -22,13 +22,37 @@ To learn about monitoring in general, see
+
+.. Set the `xpack.monitoring.collection.enabled` and 
+`xpack.monitoring.elasticsearch.collection.enabled` settings to `true` on each
+node in the cluster. 
```
Since `xpack.monitoring.elasticsearch.collection.enabled` defaults to `true`, should we ask the user to set it or would it be sufficient to ask them to verify that it's `true`, like we do above for the `xpack.monitoring.enabled` setting?

lcawl replied:
Done!

ycombinator commented on docs/reference/monitoring/configuring-monitoring.asciidoc:
```diff
@@ -119,23 +119,9 @@ xpack.monitoring.exporters:
---
+{stack-ov}/built-in-roles.html[`remote_monitoring_agent` built-in role]. 
+Alternatively, use the 
+{stack-ov}/built-in-users.html[`remote_monitoring_user` built-in user].
```
Awesome change!

lcawl commented:
@ycombinator Should I add a comment about the CCR metricset (related to https://github.com/elastic/beats/issues/8915)?

ycombinator commented:
@lcawl Thanks for checking but I think not, for now anyway. This is something we should handle in the product itself (per the discussion in https://github.com/elastic/beats/issues/8915).

lcawl commented:
Thanks, @ycombinator, I added one more piece of info about the security privileges required to update cluster settings (e.g. in https://doc-review-63bf6.firebaseapp.com/es2/configuring-metricbeat.html).  I think this is now complete.

lcawl commented:
Verified success of local gradle build.
